resources/assets/lib/beatmapsets-show/metadata-editor.tsx

```diff
@@ -154,4 +174,17 @@ export default class MetadataEditor extends React.PureComponent<Props, State> {
   private setNsfw = (e: React.ChangeEvent<HTMLInputElement>) => {
     this.setState({ nsfw: e.currentTarget.checked });
   };
+
+  private setOffset = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
+
+    // workaround for negative numbers
+    if (value === '' || value === '-') {
+      this.setState({ offset: value });
+    } else if (isNaN(parseInt(value, 10))) {
+      this.setState({ offset: '' });
+    } else {
+      this.setState({ offset: parseInt(value, 10) });
+    }
```

Comment:
No need to reset/delete content on invalid input (or even casting it to number).

```
if (/^-?\d*$/.test(value)) {
  this.setState({ offset: value })
}
```